### PR TITLE
Use standard terminology

### PR DESCRIPTION
**Pull request by ltratt**
There are no functional changes here, but this PR fixes some (unnecessary) terminological deviations, and updates some documentation.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/jit_ir/mod.rs [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
-//!   1. We use vectors rather than direct references. This allows us to squeeze the amount of the
-//!      memory used down (and also bypasses issues with representing graph structures in Rust, but
-//!      that's slightly accidental).
+//! The most important part of the IR is the *instruction* sequence created by [trace_builder].
+//! Instructions are defined by the [Inst] enum.
 //!
-//!   2. The IR is designed to allow common manipulations to be performed without requiring
-//!      reallocation.
+//! Instructions are in [SSA form](https://en.wikipedia.org/wiki/Static_single-assignment_form).
+//! Since traces are linear, we do not have explicit Φ nodes in the IR. However, there are implicit
```
[the remark this answers is not in the document]
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir/mod.rs:9]: I was doing what the SSA book is (*most* of the changes in this PR are from that book).

File: ykrt/src/compile/jitc_yk/jit_ir/mod.rs [hunk 2]
```diff
@@ -1,45 +1,56 @@
...
+//!     %7: %6
+//!     %8: add %2, %7
+//!     ```
+//!     will be displayed *deproxied* as simply `%8: add %2, %6`. Note that this is
+//!     deliberately indistinguishable from IR which directly stores `%8: add %2, %6`.
+//!
+//!   * `Tombstone` which represents an instruction which is no longer used. These are not
+//!     displayed at all.
+//!
+//! Formally speaking, deconsting.deproxying allows us to efficiently encode union sets: it is
```
Comment (ptersilie) [inline ykrt/src/compile/jitc_yk/jit_ir/mod.rs:48]: s/deconsting.deproxying/deconsting/deproxying

File: ykrt/src/compile/jitc_yk/jit_ir/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir/mod.rs:9]: Ideally we'd be consistent, but I don't think it's a big deal in this case.

**Comment by ptersilie** [comment]
Please squash.

**Comment by vext01** [comment]
Needs fmt also. You can force push if you want.

**Comment by ltratt** [comment]
Force pushed `cargo fmt`.

**Pull request merged**